src/Transform/ONNX/ConstProp.td

```diff
@@ -440,16 +452,63 @@ def DivOnesOnRhs : Pat<
       (ValuesHaveSameType $result, $x)
     ]>;
 
-// Constant Propagation for Equal
+//===----------------------------------------------------------------------===//
+// Constant propagate ONNXEqualOp
+//===----------------------------------------------------------------------===//
+
 def EqualConstProp : Pat<
     // From equal(c1, c2).
-    (ONNXEqualOp:$equalOp (ONNXConstantOp:$lhs $_, $_, $_, $_, $_, $_, $_, $_),
-                          (ONNXConstantOp:$rhs $_, $_, $_, $_, $_, $_, $_, $_)),
+    (ONNXEqualOp:$equalOp
+      (ONNXConstantOp:$lhs $_, $_, $_, $_, $_, $_, $_, $_),
+      (ONNXConstantOp:$rhs $_, $_, $_, $_, $_, $_, $_, $_)),
     // To c1 == c2
     (CreateEqualOfTwoConst $equalOp, $lhs, $rhs),
     // constraints
     [(IsFromDenseONNXConstantOp:$lhs), (IsFromDenseONNXConstantOp:$rhs)]>;
 
+//===----------------------------------------------------------------------===//
+// Constant propagate ONNXLessOp
+//===----------------------------------------------------------------------===//
+
+def LessConstPropPattern: Pat<
```

Comment:
For `Equal`, `Less` and `Greater` I don't think you need the `$result` for the builder, because of these lines https://github.com/onnx/onnx-mlir/blob/1278d70de9c03fdeebe8298c0c8c1c474764835c/utils/gen_onnx_mlir.py#L506C1-L509

But I understand why it might be worth leaving as is for now, until `GreaterOrEqual` and `LessOrEqual` are added to that list. (I can do that in a separate PR).

Reply:
good point

ok, I'll leave as is for now